src/openrct2/world/MapGen.cpp

```diff
@@ -871,11 +875,12 @@ void mapgen_generate_from_heightmap(mapgen_settings* settings)
         while (true)
         {
             uint32_t numTilesChanged = 0;
-            for (uint32_t y = 1; y <= _heightMapData.height; y++)
+            for (uint32_t y = 0; y < _heightMapData.height; y++)
```

Comment:
This shift the x and y values by 1. Isn’t that a problem? (Or alternatively, does this fix one?)

Reply:
This new implementation is more correct. This loop goes over the pixels in `_heightMapData`, and `MapgenHeightmapCoordToTileCoordsXY` returns the corresponding tile coord for these pixels. The previous implementation did this conversion implicitly.